### DISCUSSION TITLE
PUPPETEER_EXECUTABLE_PATH: Chromium executable not found | OS X

### DISCUSSION THREAD

**Discussion by ryanschieber** [jef/streetmerchant#2293] [Help] [answered]
Hi, 

after setting up my dotenv file on OSX 11.2.3 and executing npm run start, I get the following error:

`[19:05:53] error :: ✖ something bad happened, resetting streetmerchant in 5 seconds Tried to use PUPPETEER_EXECUTABLE_PATH env variable to launch browser but did not find any executable at: /snap/bin/chromium {
  "stack": "Error: Tried to use PUPPETEER_EXECUTABLE_PATH env variable to launch browser but did not find any executable at: /snap/bin/chromium\n    at ChromeLauncher.launch (/Users/mism/Downloads/streetmerchant/node_modules/puppeteer/lib/cjs/puppeteer/node/Launcher.js:80:27)\n    at async main (/Users/mism/Downloads/streetmerchant/build/src/index.js:72:15)\n    at async Timeout.loopMain [as _onTimeout] (/Users/mism/Downloads/streetmerchant/build/src/index.js:109:9)"
}`

What I've tried so far:

- Switching from node 15 to 14 and vice versa
- Downloading Chromium manually
- Editing my dotenv file to include
`PUPPETEER_EXECUTABLE_PATH=/Applications/Chromium.app`
- Editing Launcher.js to
`chromeExecutable = '/Applications/Chromium.app;`
- deleted everything and ran
`git clone https://github.com/jef/streetmerchant.git
cd streetmerchant && npm i && npm run start`again

Can anyone please give me a hint on what I'm doing wrong? I'm running out of ideas. I was also wondering, how can I add a different browser like Microsoft Edge (Chromium)?

Please see the contents of my dotenv file below:

[https://pastebin.com/N7n3F6Mv](https://pastebin.com/N7n3F6Mv)

**Comment by jef** [comment, marked as answer]
`node node_modules/puppeteer/install.js`

**Reply by ryanschieber** [reply]
Thank you for your help. I could install Chromium accordingly.

![image](https://user-images.githubusercontent.com/81982219/113629322-8fe65680-9666-11eb-9077-419812d8614a.png)

Unfortunately, I still get the same error message.

![image](https://user-images.githubusercontent.com/81982219/113625690-89a1ab80-9661-11eb-8653-78e9b552323c.png)

**Reply by jef** [reply]
Do you have the `PUPPETEER_EXECUTABLE_PATH` or other environments set? I would remove that environment now after running that command.

**Reply by ryanschieber** [reply]
What exactly should I remove? If you're referring to `PUPPETEER_EXECUTABLE_PATH` in dotenv, I removed that line. I'm not aware of any other set `PUPPETEER_EXECUTABLE_PATH` environment.

**Reply by ryanschieber** [reply]
After a clean setup, I could resolve the issue. Thank you for your help!

**Reply by jef** [reply]
Glad it works!
